src/input_formatter/expression_formatter.ts

```diff
@@ -0,0 +1,147 @@
+import { addIssue } from "../issueHandling/issueMapping";
+const allowedExpressionToken = /(bracketLeft|bracketRight|equivalence|implicationRight|implicationLeft|negation|conjunction|disjunction|equal|addition|subtraction|division|multiplication)+/;
+const logicConnector = /(conjunction|disjunction|equivalence|implicationRight|implicationLeft)+/;
+const nonLogicToken = /(bracketLeft|bracketRight|negation)+/;
```

Comment:
Why is negation considered a nonLogicToken? I suggest you add negation to your logicConnector, even though it's a unary prediacte. Then a more suitable name for nonLogicToken would maybe be brackets